casadm/cas_lib.c

```diff
@@ -1658,6 +1658,13 @@ static int get_dev_link(const char* disk, char* buf, size_t num)
 		return FAILURE;
 	}
 
+	/* check if device path isn't already device-by-id path */
+	if (!strncmp(disk, dev_by_id_dir, strnlen_s(dev_by_id_dir, sizeof(dev_by_id_dir)))) {
```

Comment:
you must sanitize @disk path somehow before attempting to compare paths as strings. canonicalize_file_name  function looks like a good place to start.